### PR TITLE
sass-mq v4: Sort added breakpoints

### LINKED ISSUE #101: Sort mq-breakpoints when adding breakpoint
When adding a breakpoint with the `mq-add-breakpoint` mixin, can the function sort the breakpoint list. This prevents libraries that loop over the mq-breakpoints from having breakpoints added later overtaking larger breakpoints.
For instance, in my code I added a phablet breakpoint. It put it on the end of the list so when inuitcss generates the width utilities the phablet one is generated later than the tablet one so it overrides it when I want different widths at the two points but only one shows up.

### PR DESCRIPTION
**Pull request by kaelig**
Fixes #101 - Thanks to @solleer for raising this issue!

## This pull request is the new sass-mq v4

This PR contains a small breaking change that **should only affect a few users**.

**⭐️ Upgrading to v4.0.0 is recommended ⭐️**

How to test v4.0.0 beta in your project:

```
npm install sass-mq@beta

# or, if you use yarn
yarn add sass-mq@beta

# and, if you use bower
bower install sass-mq@4.0.0-beta.2
```

You can also [download the latest _mq.scss](https://github.com/sass-mq/sass-mq/blob/v4.0.0-beta.2/_mq.scss) into your project.

## Before

New breakpoints added via `mq-add-breakpoint` are simply appended to `$mq-breakpoints`:

```scss
$mq-breakpoints: (
    mobile: 300px,
    tablet: 620px
);

@import 'mq';

@include mq-add-breakpoint(tiny, 160px);
@include mq-add-breakpoint(medium, 500px);

// Results in:
// $mq-breakpoints: (
//     mobile: 300px,
//     tablet: 620px,
//     tiny:   160px, < the smallest breakpoint should be first
//     medium: 500px < this breakpoint should be third
// );
```

## After

New breakpoints added via `mq-add-breakpoint` keeps `$mq-breakpoints` ordered from the smallest to the largest breakpoint:

```scss
$mq-breakpoints: (
    mobile: 300px,
    tablet: 620px
);

@import 'mq';

@include mq-add-breakpoint(tiny, 160px);
@include mq-add-breakpoint(medium, 500px);

// Results in:
// $mq-breakpoints: (
//     tiny:   160px,
//     mobile: 300px,
//     medium: 500px,
//     tablet: 620px
// );
```

Thanks to @snugug for [providing a quick-sort function](https://css-tricks.com/snippets/sass/sorting-function/)!

### CODE REVIEW

**Comment by kaelig** [comment]
@sndrs this fix is a breaking change that could have an impact on theguardian.com, as the `mq-add-breakpoint` mixin is used in those places:

https://github.com/guardian/frontend/blob/master/static/src/stylesheets/module/nav/_navigation.scss#L44
https://github.com/guardian/frontend/blob/master/static/src/stylesheets/module/identity/_disc.scss
https://github.com/guardian/frontend/blob/master/static/src/stylesheets/module/facia/_mixins.scss
https://github.com/guardian/frontend/blob/master/static/src/stylesheets/module/commercial/_frame.scss

I published `sass-mq@4.0.0-beta.1` to the npm registry, and I'd love some real-world tests before this goes to an actual 4.0.0! Let me know how I can help.

**Comment by gustavpursche** [comment]
@kaelig Thanks for thinking about backwards compatibility! I'll check the consequence, of the change, for the guardian in the following days ...

**Comment by kaelig** [comment]
@gustavpursche Thanks, any chance you may have had time to do some testing?

**Comment by sndrs** [comment]
hi @kaelig sorry i completely missed this in my notifications, apologies!

just tried it out and compile failed with `Function mq-get-breakpoint-width did not return a value` `on line 182 of node_modules/sass-mq/_mq.scss`

**Comment by kaelig** [comment]
@sndrs thanks for testing!

I got a Guardian codebase running and found the issue.

Breakpoints `containerWidestMobile` and `phablet` both are set `660px`. My PR is mistakenly de-duping them and only keeps `phablet`.

Next steps are to add this use-case to the tests and rework the sorting algorithm.
I'll get back to you when that's solved.

**Comment by kaelig** [comment]
I'm wondering if sass-mq should throw a useful error if two breakpoints have the same value (which then removes the ability for component-specific breakpoints that have the same width - so that may not be good)… thoughts?

**Comment by sndrs** [comment]
Maybe a warning?

**Comment by kaelig** [comment]
Also an option - yeah

**Comment by kaelig** [comment]
@sndrs I've pushed `4.0.0-beta.2`, which should fix the problem. Let me know how this goes for you! (here's the change: https://github.com/sass-mq/sass-mq/pull/103/commits/9fba95adec3986a980fd1bcd4420cbdbd2acf2da)

**Comment by sndrs** [comment]
there's no change in the final css diffing a build using `4.0.0-beta.2` with master build, only in sourcemaps (which would be expected) – looks good to me?

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 language: node_js
 node_js:
-  - "0.12"
   - "4"
   - "6"
+  - "8"
```
Comment (sndrs) [inline .travis.yml]: 9 too?

Comment (kaelig) [reply .travis.yml]: We could indeed! I was going for LTS support but let's add 9 :)

**Comment by kaelig** [comment]
> there's no change in the final css diffing a build using 4.0.0-beta.2 with master build, only in sourcemaps (which would be expected) – looks good to me?

Brilliant! I'll cut a v4 release very soon.

**Comment by kaelig** [comment]
@solleer would you like to give it a go in your project?

**Comment by solleer** [comment]
It seems good

**Pull request merged**